internal/characters/lauma/charge.go

```diff
@@ -0,0 +1,132 @@
+package lauma
+
+import (
+	"github.com/genshinsim/gcsim/internal/frames"
+	"github.com/genshinsim/gcsim/pkg/core/action"
+	"github.com/genshinsim/gcsim/pkg/core/attacks"
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/event"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/info"
+)
+
+var (
+	deerFrames   []int
+	chargeFrames []int
+)
+
+const (
+	chargeHitmark   = 73
+	chargeDeerFrame = 67
+	deerStatusKey   = "lauma-deer-state"
+)
+
+func init() {
+	// deer state frames
+	deerFrames = frames.InitAbilSlice(64) // skill
+	deerFrames[action.ActionAttack] = 61
+	deerFrames[action.ActionCharge] = 63
+	// deerFrames[action.ActionSkillHoldFramesOnly] = 63
+	deerFrames[action.ActionBurst] = 63
+	deerFrames[action.ActionSkill] = 63
+	deerFrames[action.ActionSwap] = 0
```

Comment:
Cancel frame shouldn't be before the transformation frame. Use `chargeDeerFrame`